packages/next/src/server/web/spec-extension/revalidate.ts

```diff
@@ -107,7 +107,11 @@ function revalidate(tags: string[], expression: string) {
       )
     }
 
-    if (workUnitStore.type === 'prerender') {
+    if (
+      workUnitStore.type === 'prerender' ||
+      // TODO we should disallow calling this from client components
+      workUnitStore.type === 'prerender-client'
```

Comment:
We do emit a build error for `revalidatePath`/`revalidateTag` in client components.

Reply:
addressed